### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by lollipopman**
Remove unused dependencies and move requires closer to where they are needed.

This request depends on pull requests #18 & #19 & #20

### CODE REVIEW

File: lib/litmus_paper/dependency/script.rb [hunk 1]
```diff
@@ -9,23 +11,27 @@ def initialize(command, options = {})
...
-            script_stderr = stderr.read.strip
-          end
-          unless script_status.success?
-            LitmusPaper.logger.info("Available check to #{@command} failed with status #{$CHILD_STATUS.exitstatus}")
-            LitmusPaper.logger.info("Failed stdout: #{script_stdout}")
-            LitmusPaper.logger.info("Failed stderr: #{script_stderr}")
+        script_status = script_stdout = script_stderr = nil
+        Open3.popen3(@command, :pgroup=>true) do |stdin, stdout, stderr, wait_thr|
+          @script_pid = wait_thr.pid
+          thstderr = Thread.new { stderr.read }
```
Comment (bjhaid) [inline lib/litmus_paper/dependency/script.rb]: why are you spawning a thread to read from `std*`, `!wait_thr.join(@timeout)` should ensure the other end of the `std*` pipes are closed so `std*` doesn't block indefinitely, if it blocks indefinitely `thstd*.value` will also block, so the thread is not really helping

File: lib/litmus_paper/dependency/script.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+require 'open3'
```
Comment (bjhaid) [inline lib/litmus_paper/dependency/script.rb]: if the intent is to remove deps you can call `Kernel.spawn` directly and create 2 pipes one for `stderr` and one for `stdout` which is what `Open3.popen*` functions do

File: spec/litmus_paper/dependency/script_spec.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@
     end
 
     it "is false when the script returns 1" do
-      check = LitmusPaper::Dependency::Script.new("false")
+      check = LitmusPaper::Dependency::Script.new("echo Hello && echo Goodbye 1>&2 && false")
```
Comment (bjhaid) [inline spec/litmus_paper/dependency/script_spec.rb]: Any reason why the behavior of this spec is changing?

File: lib/litmus_paper/dependency/script.rb (hunk 2 above)
Comment (lollipopman) [inline lib/litmus_paper/dependency/script.rb]: I am trying to remove libraries outside of the standard library

File: spec/litmus_paper/dependency/script_spec.rb (hunk 3 above)
Comment (lollipopman) [inline spec/litmus_paper/dependency/script_spec.rb]: good call out, this was to ensure stdout and stderr are being outputted correctly in the logs

Comment (bjhaid) [inline spec/litmus_paper/dependency/script_spec.rb]: maybe add a test to make this explicit?

**Pull request merged**